crates/sui-framework/packages/deepbook/sources/clob_v3.move

```diff
@@ -0,0 +1,3043 @@
+        maker_address: address,
+        original_quantity: u64,
+        base_asset_quantity_filled: u64,
+        base_asset_quantity_remaining: u64,
+        price: u64,
+        taker_commission: u64,
+        maker_rebates: u64
+    }
+
+    /// Emitted when user deposit asset to custodian
+    struct DepositAsset<phantom Asset> has copy, store, drop {
+        /// object id of the pool that asset deposit to
+        pool_id: ID,
+        /// quantity of the asset deposited
+        quantity: u64,
+        /// owner address of the `AccountCap` that deposit the asset
+        owner: address
+    }
+
+    /// Emitted when user withdraw asset from custodian
+    struct WithdrawAsset<phantom Asset> has copy, store, drop {
+        /// object id of the pool that asset withdraw from
+        pool_id: ID,
+        /// quantity of the asset user withdrew
+        quantity: u64,
+        /// owner ID of the `AccountCap` that withdrew the asset
+        owner: address
+    }
+    // <<<<<<<<<<<<<<<<<<<<<<<< Events <<<<<<<<<<<<<<<<<<<<<<<<
+
+    struct Order has store, drop {
+        // For each pool, order id is incremental and unique for each opening order.
+        // Orders that are submitted earlier has lower order ids.
+        // 64 bits are sufficient for order ids whereas 32 bits are not.
+        // Assuming a maximum TPS of 100K/s of Sui chain, it would take (1<<63) / 100000 / 3600 / 24 / 365 = 2924712 years to reach the full capacity.
+        // The highest bit of the order id is used to denote the order type, 0 for bid, 1 for ask.
+        order_id: u128,
+        client_order_id: u128,
+        // Only used for limit orders.
+        price: u64,
+        // quantity when the order first placed in
+        original_quantity: u64,
+        // quantity of the order currently held
+        quantity: u64,
+        is_bid: bool,
+        /// Order can only be canceled by the `AccountCap` with this owner ID
+        owner: address,
+        // Expiration timestamp in ms.
+        expire_timestamp: u64,
+        // reserved field for prevent self_matching
+        self_matching_prevention: u8,
+        // reserved field for a tombstoned order. If true means that this order is dead and will not be used.
+        is_tombstoned: bool
+    }
+
+    struct TickLevel has store {
+        price: u64,
+        // The index in the queue is the order_id.
+        order_queue: BigQueue<Order>,
+        // Count of bid elements popped from queue.
+        count_elements_popped: u64,
+        // Next portion of the 64 bit order id starts at 0 for bids, and 1<<63 for asks
+        next_order_id: u64
+    }
+
+    struct Pool<phantom BaseAsset, phantom QuoteAsset> has key, store {
+        // The key to the following Critbit Tree are order prices.
+        id: UID,
+        // All open bid orders.
+        bids: CritbitTree<TickLevel>,
+        // All open ask orders.
+        asks: CritbitTree<TickLevel>,
```

Comment:
tick levels can't overlap. The one tree implementation doesn't work well because then getting the max node and min node would be weird. Whereas right now it's every easy to ask for max and min of the orders

Reply:
I think I understand -- currently you can easily look for the min/max bid/ask independently, but if you put bids and asks in the same tree, your query would give you the min bid or ask or the max bid or ask, and you would potentially have to linearly scan to find a bid if you first found an ask or vice versa?